### DISCUSSION TITLE
"luma.gl: Device pixel ratio clamped"

### DISCUSSION THREAD

**Discussion by richak8** [visgl/deck.gl#6786] [Q&A]
Browser version:  Chrome (Version 99.0.4844.83)
Frameworks: [deck.gl/core](http://deck.gl/core):  8.2.8, [deck.gl/google-maps](http://deck.gl/google-maps):  8.3.0.
OS:  macOS catalina (version: 10.15.7)
can anyone please help me with this issue?

**Comment by kylebarron** [comment]
As @Pessimistress replied in Slack, that's not an error, just a warning. Can you share your code?

**Reply by ghost** [reply]
Why does it happen (why is this clamping necessary) and how to avoid it?
I'm trying to render at 10x resolution to produce a video, but I'm getting this warning and the value gets clamped to about 5x.

**Reply by Pessimistress** [reply]
There is a limit to how big the WebGLContext will allow you to set the drawing buffer. We try to set it to the size that matches your resolution, but if that fails, we'll have to fall back to a lower resolution ("clamped").
